include/fmt/core.h

```diff
@@ -2718,11 +2711,15 @@ FMT_CONSTEXPR FMT_INLINE void parse_format_string(
       if (from == to) return;
       for (;;) {
         const Char* p = nullptr;
-        if (!find<IS_CONSTEXPR>(from, to, Char('}'), p))
-          return handler_.on_text(from, to);
+        if (!find<IS_CONSTEXPR>(from, to, Char('}'), p)) {
+          handler_.on_text(from, to);
+          return;
```

Comment:
These and changes below look unnecessary, please revert.

Reply:
I commented on the #3204 reasons for this change and 2 more below.

Reply:
Thanks for providing more context but I still don't see reasons for changing this. Let's revert. The other changes seem fine.